src/components/cuda/cupti_profiler.c

```diff
@@ -0,0 +1,1576 @@
+            return PAPI_EMISC);
+
+        NVPW_MetricsContext_GetMetricNames_Begin_Params getMetricNameBeginParams = {
+            .structSize = NVPW_MetricsContext_GetMetricNames_Begin_Params_STRUCT_SIZE,
+            .pPriv = NULL,
+            .pMetricsContext = avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+            .hidePeakSubMetrics = !listsubmetrics,
+            .hidePerCycleSubMetrics = !listsubmetrics,
+            .hidePctOfPeakSubMetrics = !listsubmetrics,
+        };
+        NVPW_CALL(NVPW_MetricsContext_GetMetricNames_BeginPtr(&getMetricNameBeginParams), return PAPI_EMISC);
+
+        avail_events[gpu_id].num_metrics = getMetricNameBeginParams.numMetrics;
+        avail_events[gpu_id].nv_metrics = initialize_dynamic_event_list_size(avail_events[gpu_id].num_metrics);
+        if (avail_events[gpu_id].nv_metrics == NULL) {
+            errno = PAPI_ENOMEM;
+            goto fn_exit;
+        }
+        for (i=0; i<avail_events[gpu_id].num_metrics; i++) {
+            errno = insert_event_record(avail_events[gpu_id].nv_metrics,
+                                      getMetricNameBeginParams.ppMetricNames[i],
+                                      i, 0);
+            if (errno != PAPI_OK)
+                goto fn_exit;
+        }
+
+        NVPW_MetricsContext_GetMetricNames_End_Params getMetricNameEndParams = {
+            .structSize = NVPW_MetricsContext_GetMetricNames_End_Params_STRUCT_SIZE,
+            .pPriv = NULL,
+            .pMetricsContext = avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+        };
+        NVPW_CALL(NVPW_MetricsContext_GetMetricNames_EndPtr((NVPW_MetricsContext_GetMetricNames_End_Params *) &getMetricNameEndParams), return PAPI_EMISC);
+
+    }
+    TOCK;
+    TIMEDBG("Time to get all metric names =");
+    TICK;
+    char evt_name[PAPI_2MAX_STR_LEN];
+    event_rec_t *find=NULL;
+    int len;
+    for (gpu_id=0; gpu_id<num_gpus; gpu_id++) {
+        for (i=0; i<avail_events[gpu_id].num_metrics; i++) {
+            len = snprintf(evt_name, PAPI_2MAX_STR_LEN, "%s:device=%d", avail_events[gpu_id].nv_metrics->evts[i].name, gpu_id);
+            if (len > PAPI_2MAX_STR_LEN) {
+                ERRDBG("String formatting exceeded maximum length.\n");
+                errno = PAPI_ENOMEM;
+                goto fn_exit;
+            }
+            if (find_event_name(all_evt_names, evt_name, &find) == PAPI_ENOEVNT) {
+                errno = insert_event_record(all_evt_names, evt_name, i, 0);
+                if (errno != PAPI_OK) {
+                    goto fn_exit;
+                }
+            }
+            // LOGDBG("Name:\t\t%s\nDescription:\t%s\n\n",
+            //    all_evt_names->evts[all_evt_names->count-1].name, avail_events[gpu_id].nv_metrics->evts[i].desc);
+        }
+    }
+    TOCK;
+    TIMEDBG("Time to transfer all metric names to component =");
```

Comment:
I feel you have too many log messages.

Reply:
There should be enough log messages, so that if something fails for the user, the log messages tell us where and what the problem was, and how we can fix it.